server/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java

```diff
@@ -143,13 +156,16 @@ public ShardUpsertRequest(StreamInput in) throws IOException {
         } else {
             isRetry = false;
         }
+        if (in.getVersion().onOrAfter(Version.V_5_2_0)) {
```

Comment:
As @BaurzhanSakhariev already noted on the original commit, this is not optimal to roll-out as for the `5.1` branch we have to change it to the latest `5.1` snapshot version and after released have to adjust it on master again.
Also to note that streaming the stage may not really needed as it is only taken into account while streaming out the request on the primary to the replica. So streaming of it is not needed for the fix, but for consistency I thought it's better to stream it.

Reply:
Could we change it already to V_5_1_2 on master?